daemon/container_operations.go

```diff
@@ -691,6 +691,20 @@ func (daemon *Daemon) connectToNetwork(cfg *config.Config, container *container.
 	}
 	nwName := n.Name()
 
+	// If idOrName is a partial ID, we replace it with the actual network name to avoid duplicated entries in
+	// `container.NetworkSettings.Networks`.
+	if idOrName != n.ID() && idOrName != nwName {
+		endpointConfigState, ok := container.NetworkSettings.Networks[idOrName]
+		if ok {
+			delete(container.NetworkSettings.Networks, idOrName)
+		} else {
```

Comment:
Should this be an error condition though?

```bash
docker network create foo
66ed1fdabd817bd32dbb181a4b2d18e391f8fe9115f216c008deb1df21c30cb0

docker run -d --name mycontainer --network=foo nginx:alpine
065994be656c98c957b5206b5cf594b712f4abef70b46e85573f4f9a572a19c1

docker network connect foo mycontainer
Error response from daemon: endpoint with name mycontainer already exists in network foo
```

Reply:
I simplified this condition because much of it wasn't needed. However I'm not sure what you mean in your previous comment :thinking:

Reply:
So, you shared privately:

> So, my thinking there was that with the change you made would overwrite previous settings if the NetworkSettings contains duplicates (e.g. same network once connected by ID and onces by Name (or once by partial ID.
In that case “which ever came last” would remove the previous setting, and replace it.

I simplified this change to not overwrite the `EndpointSettings` entry since few lines below the entry is created _iff_ the `CreateEndpoint` call succeed. This is the function that return the error `endpoint with name mycontainer already exists in network foo`. So we should be good now.

Like for my other PR, it doesn't feel right to handle such data normalization in `connectToNetwork` but this can't be handled before the `findAndAttachNetwork` call has been made, because of swarm-scope networks.